File: bindings/test_bindings.py
import os
import sys
from sandbox import LibMountSandbox

def main():
    print("Testing Python FFI Bindings...")
    
    if sys.platform == 'darwin':
        lib_name = 'libmountsandbox.dylib'
    elif sys.platform == 'win32' or sys.platform == 'cygwin':
        lib_name = 'mountsandbox.dll'
    else:
        lib_name = 'libmountsandbox.so'

    # Resolve relative to current file
    lib_path = os.path.join(os.path.dirname(__file__), "..", "build", lib_name)
    sandbox = LibMountSandbox(lib_path=lib_path)
    
    status = sandbox.execute("dummy", ["ls", "-la"])
    print(f"Command returned status: {status}")

if __name__ == "__main__":
    main()

File: bindings/test_bindings.rb
require_relative 'sandbox'
require 'rbconfig'

def main
  puts "Testing Ruby FFI Bindings..."
  
  lib_name = case RbConfig::CONFIG['host_os']
             when /darwin/ then 'libmountsandbox.dylib'
             when /mswin|msys|mingw|cygwin|bccwin|wince|emc/ then 'mountsandbox.dll'
             else 'libmountsandbox.so'
             end
  
  # Resolve relative to current file
  lib_path = File.join(File.dirname(__FILE__), "..", "build", lib_name)
  
  # Try to execute ls
  sandbox = LibMountSandbox::SandboxWrapper.new(lib_path)
  
  status = sandbox.execute("dummy", ["echo", "hello"])
  puts "Command returned status: #{status}"
end

if __FILE__ == $0
  main
end
